hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/OAuth2AuthProvider.java

```diff
@@ -0,0 +1,65 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.http;
+
+import pl.allegro.tech.hermes.api.OAuth2AuthenticationData;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.common.cache.zookeeper.NodeCache;
+import pl.allegro.tech.hermes.consumers.consumer.ConsumerMessageSender;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import com.github.scribejava.core.model.OAuthConfig;
+
+public class OAuth2AuthProvider implements HttpAuthorizationProvider {
```

Comment:
i wonder if it is a good idea to run this blocking code in Consumer thread.. this is also in regard to "what to do when all messages need to authorize" comment below. I think we could go with sth like separate thread pool to execute oauth requests, isolating it form main consumer loop, and this would "communicate" via some thread safe collection (ConcurrentHashMap?) with the sender itself. Also we need to think about some rate limiting for OAuth service - we deal with some pretty serious traffic here, i don't think OAuth gateways are prepared to get hit by hundreds of rps.